pkg/controller/master/virtualmachine/vmi_network_controller.go

```diff
@@ -60,56 +54,23 @@ func (h *VMNetworkController) updateVMDefaultNetworkMacAddress(vmi *kubevirtv1.V
 			vmiInterfaces[iface.Name] = iface.MAC
 		}
 	}
-
-	for i, vmIface := range vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces {
-		macAddress, ok := vmiInterfaces[vmIface.Name]
-		// only set the network mac address when it has no existing value
-		if ok && vmIface.MacAddress == "" {
-			logrus.Debugf("set VM %s management network %s macAddress to %s", vm.Name, vmIface.Name, macAddress)
-			vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces[i].MacAddress = macAddress
-		}
-	}
-
-	// note: this function is related to vm/vmi, should always run on vmi change
-	if err := h.regenerateControllerRevision(vmi, vm); err != nil {
+	vmiInterfaceBytes, err := json.Marshal(vmiInterfaces)
+	if err != nil {
 		return err
 	}
 
-	if equality.Semantic.DeepEqual(vmCopy.Spec.Template.Spec.Domain.Devices, vm.Spec.Template.Spec.Domain.Devices) {
-		return nil
-	}
-	if _, err := h.vmClient.Update(vmCopy); err != nil {
-		return err
+	if vmCopy.Annotations == nil {
+		vmCopy.Annotations = make(map[string]string)
 	}
+	vmCopy.Annotations[util.AnnotationMacAddressName] = string(vmiInterfaceBytes)
 
-	return nil
-}
-
-func (h *VMNetworkController) regenerateControllerRevision(vmi *kubevirtv1.VirtualMachineInstance, vm *kubevirtv1.VirtualMachine) error {
-	crObj, err := h.crCache.Get(vmi.Namespace, vmi.Status.VirtualMachineRevisionName)
-	if err != nil {
-		return fmt.Errorf("error fetch controller revision object for vmi %s-%s: %v", vmi.Name, vmi.Namespace, err)
-	}
-
-	revisionSpec := &vmwatch.VirtualMachineRevisionData{}
-	if err = json.Unmarshal(crObj.Data.Raw, revisionSpec); err != nil {
-		return err
-	}
-	if !equality.Semantic.DeepEqual(revisionSpec.Spec.Template, vm.Spec.Template) && vm.Generation != vm.Status.ObservedGeneration {
-		patch, patchGenError := patchVMRevision(vm)
-		if patchGenError != nil {
-			return fmt.Errorf("error during patch generation for vmi %s-%s: %v", vmi.Name, vmi.Name, patchGenError)
+	if !reflect.DeepEqual(vm, vmCopy) {
```

Comment:
in this case, compare the Annotations[util.AnnotationMacAddressName] and  string(vmiInterfaceBytes) seems to be enough